src/Metadata/Property/Factory/AnnotationSubresourceMetadataFactory.php

```diff
@@ -82,7 +82,8 @@ private function updateMetadata(ApiSubresource $annotation, PropertyMetadata $pr
         $type = $propertyMetadata->getType();
         $isCollection = $type->isCollection();
         $resourceClass = $isCollection ? $type->getCollectionValueType()->getClassName() : $type->getClassName();
+        $maxDepth = $annotation->maxDepth;
```

Comment:
Shouldn't we check that `maxDepth` is `int` here?

Reply:
Doctrine Annotation will handle that because of the PHPDock IIRC.

Reply:
Oh, I didn't know that. Nice.

![](http://i.nusphere.com/images/as/box-phpdock-20.png)